lazyflow/operators/ioOperators/opTiffReader.py

```diff
@@ -136,14 +108,13 @@ def execute(self, slot, subindex, roi, result):
                 tiff_page_ndindex = roi_page_ndindex + page_index_roi[0]
                 tiff_page_list_index = numpy.ravel_multi_index(tiff_page_ndindex, self._non_page_shape)
                 logger.debug("Reading page: {} = {}".format(tuple(tiff_page_ndindex), tiff_page_list_index))
-                page_data = vigra.impex.readImage(
-                    self._filepath, dtype="NATIVE", index=int(tiff_page_list_index), order="C"
-                )
+                with tifffile.TiffFile(self._filepath, mode="r") as f:
+                    page_data = f.series[0].asarray(key=int(tiff_page_list_index), maxworkers=1)
             else:
                 # Only a single page
-                page_data = vigra.impex.readImage(self._filepath, dtype="NATIVE", index=0, order="C")
+                with tifffile.TiffFile(self._filepath, mode="r") as f:
+                    page_data = f.series[0].asarray(maxworkers=1)
```

Comment:
```python
for roi_page_ndindex in ...:
    with tifffile.TiffFile(...) as f:
        key = None
        if self._non_page_shape:
            tiff_page_ndindex = page_index_roi[0] + roi_page_ndindex
            key = int(numpy.ravel_multi_index(tiff_page_ndindex, self._non_page_shape))
            logger.debug(...)
        page_data = f.series[0].asarray(key=key, maxworkers=1)
```

Reply:
sure! Although I don't see a reason to enter the context manager before accessing data...

Reply:
actually, In order to minimize file opening - the context manager should probably be outside the loop :)